web/src/main/java/org/springframework/security/web/server/header/CompositeServerHttpHeadersWriter.java

```diff
@@ -43,8 +42,10 @@ public CompositeServerHttpHeadersWriter(List<ServerHttpHeadersWriter> writers) {
 
 	@Override
 	public Mono<Void> writeHttpHeaders(ServerWebExchange exchange) {
-		Stream<Mono<Void>> results = writers.stream().map( writer -> writer.writeHttpHeaders(exchange));
-		return Mono.when(results.collect(Collectors.toList()));
+		List<Mono<Void>> results = new ArrayList<>();
```

Comment:
I appreciate that you used an interface as your reference type here (e.g. `List` instead of `ArrayList`).

Reply:
Thanks